controller/share_manager_controller.go

```diff
@@ -405,27 +405,51 @@ func (c *ShareManagerController) isShareManagerRequiredForVolume(sm *longhorn.Sh
 	return false
 }
 
-func (c *ShareManagerController) createShareManagerAttachmentTicket(sm *longhorn.ShareManager, va *longhorn.VolumeAttachment) {
+func (c *ShareManagerController) createShareManagerAttachmentTicket(sm *longhorn.ShareManager, va *longhorn.VolumeAttachment) error {
```

Comment:
You add `error` returned value, but forgot the update the caller needs to check the error.  

[/controller/share_manager_controller.go#L584-L586](https://github.com/longhorn/longhorn-manager/blob/a09908dab732d514a8750dd8ead776129c92838e/controller/share_manager_controller.go#L584-L586)
```go
	c.createShareManagerAttachmentTicket(sm, va)

	return nil
```

Reply:
Actually, that was kind of on purpose.  If there is an error from createShareManagerAttachmentTicket(), and it is returned back up by syncShareManagerVolume(), the current code in syncShareManager() will just skip the calls to syncShareManagerPod() and syncShareManagerEndpoint().  Is that really what we want?  It seems to open up deadlock possibilities for when successfully doing the Pod sync would clear up the situation causing an error in Volume sync, but we never get the chance to do that.

Reply:
Agreed, that is intended. However, suggest updating the code as below to make it explicit. 

```go
_ = c.createShareManagerAttachmentTicket(sm, va)
```

Reply:
Done.